public/app/features/admin/UserPermissions.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { PureComponent } from 'react';
+import { ConfirmButton } from '@grafana/ui';
+import { cx, css } from 'emotion';
+
+interface Props {
+  isGrafanaAdmin: boolean;
+
+  onGrafanaAdminChange: (isGrafanaAdmin: boolean) => void;
+}
+
+interface State {
+  isEditing: boolean;
+  currentAdminOption: string;
+}
+
+export class UserPermissions extends PureComponent<Props, State> {
+  state = {
+    isEditing: false,
+    currentAdminOption: this.props.isGrafanaAdmin ? 'YES' : 'NO',
+  };
+
+  handleChangeClick = () => {
+    this.setState({ isEditing: true });
+  };
+
+  handleCancelClick = () => {
+    this.setState({
+      isEditing: false,
+      currentAdminOption: this.props.isGrafanaAdmin ? 'YES' : 'NO',
+    });
+  };
+
+  handleGrafanaAdminChange = () => {
+    const { currentAdminOption } = this.state;
+    const newIsGrafanaAdmin = currentAdminOption === 'YES' ? true : false;
+    this.props.onGrafanaAdminChange(newIsGrafanaAdmin);
+  };
+
+  handleAdminOptionSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ currentAdminOption: event.target.value });
+  };
+
+  render() {
+    const { isGrafanaAdmin } = this.props;
+    const { isEditing, currentAdminOption } = this.state;
+    const changeButtonContainerClass = cx(
+      'pull-right',
+      css`
+        margin-right: 0.6rem;
+      `
+    );
```

Comment:
I think we should avoid importing `cx` and `emotion` and use `stylesFactory` instead?